File: pages/anketa/args.pl
#!/usr/bin/env perl

use strict;
use Getopt::Long;
use Pod::Usage;

my $help;
my $message = "";
my $verbose;

@ARGV and GetOptions(
    "h" => \$help,
    "m:s" => \$message,
    "v" => \$verbose,
) or pod2usage(1);
pod2usage(-verbose => 2, -exitval => 2) if $help;

my $count = 0;
my $arg;

foreach $arg (@ARGV) {
    print($arg . "\n") if $verbose;
    $count++;
}

if($message ne "") {
    print($message . "\n");
}

print($count . "\n");

__END__

=head1 NAME

    printargs.pl - Print the number of arguments.

=head1 SYNOPSIS

    printargs.pl [options] [arguments]

=head1 OPTIONS

    -h      Show help message.

    -m MSG  Specify a custom message.


=head1 EXAMPLE

    printargs.sh a b c

    printargs.pl -m "Arguments count: " a b c

    printargs.pl -v a b c

    printargs.pl -h

File: pages/anketa/args.py
#!/usr/bin/env python

import argparse

parser = argparse.ArgumentParser(description='Print the number of arguments.')
parser.add_argument('arguments', metavar='ARG', type=str, nargs='*', help='some arguments')
parser.add_argument('-m', dest='message', default='', help='custom message')
parser.add_argument('-v', '--verbose', action='store_true', dest='verbose', help='verbose mode')

args = parser.parse_args()

count = 0

for a in args.arguments:
    if args.verbose:
        print a
    count += 1

if args.message != '':
    print(args.message)

print(count)
